core/app/services/workarea/copy_page.rb

```diff
@@ -0,0 +1,31 @@
+module Workarea
+  class CopyPage
+    def initialize(page, attrs = {})
+      @page = page
+      @attributes = Workarea.config.page_copy_default_attributes.merge(attrs)
+    end
+
+    def perform
```

Comment:
There's a bunch going on in this method, let's refactor it down into some smaller methods like `copied_from`, `copy_page`, `copy_content`. For inspiration, check out `CopyOrder`.